tools/windows/DatadogAgentInstaller/CustomActions/UserCustomActions.cs

```diff
@@ -21,35 +23,37 @@ public class UserCustomActions
         private readonly ISession _session;
         private readonly INativeMethods _nativeMethods;
         private readonly IRegistryServices _registryServices;
-        private readonly IDirectoryServices _directoryServices;
-        private readonly IFileServices _fileServices;
+        private readonly IFileSystemServices _fileSystemServices;
         private readonly IServiceController _serviceController;
 
+        // Created as needed on a per-customaction basis
+        private RollbackDataStore rollbackDataStore;
```

Comment:
split up `UserCustomActions` -> [`ConfiguerUserCustomActions`, `ProcessUserCustomActions`]. left `ConfigureUser`+`UninstallUser` together because `UninstallUser` is meant to undo what `ConfigureUser` does.